mod/wikirate_source/set/type/source/source.rb

```diff
@@ -36,26 +38,20 @@ def indirect_contributor_search_args
 end
 
 def has_file_or_text?
-  file_card = subcards["+File"]
-  text_card = subcards["+Text"]
-  ( file_card && (file_card.stringify_keys.has_key?("file") || file_card.stringify_keys.has_key?("remote_file_url"))) || 
-  ( text_card && ( text_card_content = (text_card.stringify_keys)["content"] ) && !text_card_content.empty? )
+  file_card = subfield(:file)
+  text_card = subfield(:text)
+  ( file_card && file_card.attachment.present? ) || 
+  ( text_card && text_card.content.present? )
 end
 
-event :process_source_url, :before=>:process_subcards, :on=>:create, :when=>proc{  |c| !c.has_file_or_text? } do
-
-  linkparams = subcards["+#{ Card[:wikirate_link].name }"]
-  url = linkparams && linkparams[:content] or errors.add(:link, " does not exist.")
+event :process_source_url, :before=>:process_subcards, :on=>:create, 
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Surrounding space missing for operator `=>`.<br>Trailing whitespace detected.